chainermn/datasets/scatter.py

```diff
@@ -33,6 +34,8 @@ def scatter_dataset(dataset, comm, root=0, shuffle=False,
             If ``None``, the permutation is changed randomly.
         max_buf_len (int): Max buffer size to be used at broadcasting
             binaries. Must not be larger than 2147483647.
+        strict (bool): Strictly scatters the index and avoid data duplication
```

Comment:
I guess we need to document what happens when `len(dataset) % comm.size != 0`. 

The reason why the previous `scatter_dataset()` allows duplication is that all processes must have the same number of samples to keep the iteration count correct.  I'm not 100% sure, but using `strict` option comes with a cost of breaking the restriction ?

Reply:
Updated the document.